torch/nn/modules/container.py

```diff
@@ -425,18 +423,14 @@ def forward(self, x):
                 return x
     """
 
-    _parameters: Dict[str, 'Parameter']  # type: ignore[assignment]
-
-    def __init__(self, parameters: Optional[Iterable['Parameter']] = None) -> None:
+    def __init__(self, values: Optional[Iterable[Any]] = None) -> None:
         super(ParameterList, self).__init__()
-        self._initialized = True
-        if parameters is not None:
-            self += parameters
-
-    def __setstate__(self, state):
-        state['_initialized'] = False
-        super(ParameterList, self).__setstate__(state)
-        self._initialized = True
+        self._size = 0
+        if values is not None:
+            # BC-breaking!
+            # Plain Tensors passed here used to be automatically converted into
+            # Parameters. Do we want to keep that behavior?
```

Comment:
As discussed externally, I think it may be good to keep this behavior in the constructor to reduce the BC-breaking impact.